### DISCUSSION TITLE
docs: Revamp documentation for better styling and readability

### DISCUSSION THREAD

**Discussion by jamesli2021** [mikro-orm/mikro-orm#6029] [Ideas]
Reflecting on my previous experience, I recall effortlessly running Mikro, but today, I felt lost while attempting to set up and pull an existing Postgres database with Mikro CLI, and gave up after a few hours.

The challenge I encountered was the confusing setup process.

Currently, I'm using Astro web framework, and I think the Starlight documentation platform could potentially enhance the clarity and visual appeal of your current documentation. What is your thought on my suggestion?

https://astro.build/
https://starlight.astro.build/

**Comment by B4nan** [comment]
Did you go through the getting started guide? What problems did you have?

I am quite happy with docusaurus, no plans to switch because of content issues, I don't see how that could help much.

**Comment by jamesli2021** [comment]
Not sure, how to resolve issue with file extension, what am I missing?


npx mikro-orm generate-entities --dump
mikro-orm generate-entities

Generate entities based on current database schema

Options:
      --config   Set path to the ORM configuration file                 [string]
  -s, --save     Saves entities to directory defined by --path         [boolean]
  -d, --dump     Dumps all entities to console                         [boolean]
  -p, --path     Sets path to directory where to save entities          [string]
      --schema   Generates entities only for given schema               [string]
  -v, --version  Show version number                                   [boolean]
  -h, --help     Show help                                             [boolean]

TypeError: Unknown file extension ".ts" for /Users/s/Documents/book2/mikro-orm.config.ts
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:160:9)
    at defaultGetFormat (node:internal/modules/esm/get_format:203:36)
    at defaultLoad (node:internal/modules/esm/load:143:22)
    at async ModuleLoader.load (node:internal/modules/esm/loader:553:7)
    at async ModuleLoader.moduleProvider (node:internal/modules/esm/loader:434:45)
    at async link (node:internal/modules/esm/module_job:87:21) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}

**Reply by B4nan** [reply]
First thing you could try is the search 

<img width="582" alt="image" src="https://github.com/user-attachments/assets/1a738b56-9bba-45fa-be87-a4b351551ef9">

or with the error code 

<img width="572" alt="image" src="https://github.com/user-attachments/assets/446e33e2-4414-4e30-ab5e-00f09da9ad31">

This points to the getting started guide.

**Comment by jamesli2021** [comment]
My Astro web framework is using TypeScript but I have not modify tsconfig.json at this point.

**Comment by jamesli2021** [comment]
If I add this and still having the same error.


```
// https://mikro-orm.io/docs/entity-generator
import { defineConfig } from '@mikro-orm/postgresql';
import { EntityGenerator } from '@mikro-orm/entity-generator';

export default defineConfig({
  dbName: 'book',
  extensions: [EntityGenerator],
  // for vitest to get around `TypeError: Unknown file extension ".ts"` (ERR_UNKNOWN_FILE_EXTENSION)
  dynamicImportProvider: id => import(id),
});
~                                                                                                                                                               
~     
```

My suggestion is it's easier for you to create Astro example?

**Reply by B4nan** [reply]
You should be using `npx mikro-orm-esm` most likely, I guess your project is ESM if you see this error.

**Comment by jamesli2021** [comment]
Updated: I hit with another issue.
```
Using the export keyword between a decorator and a class is not allowed. Please use export @dec class instead.
```

Does my example appear to be correct?
```

import { MikroORM } from '@mikro-orm/postgresql';
import { Users } from "generated-entities/Users";


const orm = await MikroORM.init({
    entitiesTs: ['./generated-entities'], // Path to your compiled entities
    dbName: 'book2',

});

console.log(orm.em); // access EntityManager via `em` property
console.log(orm.schema);


const newUser = new Users();
newUser.name = 'test 1';
newUser.email = 'test@example.com';

await orm.em.persist(newUser).flush();

```

**Reply by B4nan** [reply]
You need legacy decorators.
